gufe/protocols/protocol.py

```diff
@@ -32,8 +32,11 @@ class ProtocolResult(GufeTokenizable):
     - `get_uncertainty`
     """
 
-    def __init__(self, **data):
+    def __init__(self, n_protocol_dag_results: Optional[int] = None, **data):
         self._data = data
+        self._n_protocol_dag_results = (
+            n_protocol_dag_results if n_protocol_dag_results is not None else 0
```

Comment:
I'm trying to wrap my head around the use case here, could you explain the case where None happens? I.e. at first glance setting None to zero seems incorrect if None means that the gather method did not pass through a value. In that case None means undefined and it's much cleaner than getting 0 with the idea that you got no results (but you do).

Reply:
I think in the context of a `Protocol`'s `gather`, you'll never see a `None` passed in. I can see instances where `ProtocolResult`s are used directly, incorrectly or not, and a `None` would make sense. But of course I immediately throw away the `None` in the code you highlighted above. I suppose there is still some indecision on my part what the right signature should be and which values the `ProtocolResult` holds on to. I'm tempted to just drop `Optional[int]` and use `int` with default `0`.

Just from the tests, I see instances where someone would make a `ProtocolResult` directly and I'll probably do something similar in testing parts of `stratocaster` downstream:

```python
class DummyProtocolResult(gufe.ProtocolResult):
    def get_estimate(self):
        return self.data['estimate']

    def get_uncertainty(self):
        return self.data['uncertainty']


class TestProtocolResult(GufeTokenizableTestsMixin):
    cls = DummyProtocolResult
    key = 'DummyProtocolResult-b7b854b39c1e37feabec58b4000680a0'
    repr = f'<{key}>'

    @pytest.fixture
    def instance(self):
        return DummyProtocolResult(
            estimate=4.2 * unit.kilojoule_per_mole,
            uncertainty=0.2 * unit.kilojoule_per_mole,
        )
```

Reply:
Not necessarily. I'll be testing stratocaster strategies based on this branch until it's merged in the hopes that we get some solution that's close in spirit to what we have here.